internal/pkg/iostreams/io.go

```diff
@@ -156,7 +159,39 @@ func (s *system) ReadSecret() ([]byte, error) {
 		return nil, fmt.Errorf("prompting is disabled")
 	}
 
-	return term.ReadPassword(int(s.in.Fd()))
+	fd := int(s.in.Fd())
+
+	// Store and restore the terminal status on interruptions to
+	// avoid that the terminal remains in the password state
+	// This is necessary as for https://github.com/golang/go/issues/31180
+	oldState, err := term.GetState(fd)
+	if err != nil {
+		return nil, err
+	}
+
+	type Buffer struct {
+		Buffer []byte
+		Error  error
+	}
+	errorChannel := make(chan Buffer, 1)
+
+	// Cancelled context restores the terminal, otherwise the no-echo mode would remain intact
+	go func() {
+		<-s.ctx.Done()
+		if oldState != nil {
+			_ = term.Restore(fd, oldState)
+		}
+		errorChannel <- Buffer{Buffer: make([]byte, 0), Error: ErrInterrupt}
```

Comment:
```suggestion
	errorChannel := make(chan Buffer, 1)
	doneChannel := make(chan struct{})
	defer close(doneChannel)

	// Cancelled context restores the terminal, otherwise the no-echo mode would remain intact
	go func() {
		select {	
		case <-doneChannel:
			return
		case <-s.ctx.Done():
		}
		
		if oldState != nil {
			_ = term.Restore(fd, oldState)
		}
		errorChannel <- Buffer{Buffer: make([]byte, 0), Error: context.Cause(s.ctx)}
```

So these are nits, but suggestion two changes:

1. Add a doneChannel that causes the goroutine to exit. This is a nit, since it is a CLI and isn't expected to be long running anyways, but as it is, this leaks a goroutine until the CLI exits.
2. Lets not add the ErrInterrupt but instead use context.Cause like you had it before.

Reply:
❤️ it.